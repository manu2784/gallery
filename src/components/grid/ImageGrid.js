import React from "react";

export default class ImageGrid extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
        images: null,
        loading:true,
    }
  }

   componentDidMount(){
    fetch("https://cors-anywhere.herokuapp.com/https://pixabay.com/api/?key=6473511-0417f2cad683f1bee54cafe15")
    .then(res=>res.json())
    .then(body=>{
        this.setState({images:body.hits, loading:false});
    });
   
  }

  render(){
    console.log(this.state.loading);
    return (
     <h1>Hello</h1>
    )
  }
}
