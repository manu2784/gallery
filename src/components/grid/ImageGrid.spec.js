import React from "react";
import { shallow, configure } from "enzyme";
import ImageGrid from "./ImageGrid";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



describe("ImageGrid", () => {
  let props = {
    match:{ params: {
      id:"56546565"
    } }
  }

  it("should render the image grid page", () => {

    const grid = shallow(<ImageGrid />);
    expect(grid).toMatchSnapshot();
  });

});


