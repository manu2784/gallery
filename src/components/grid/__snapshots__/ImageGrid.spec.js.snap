// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGrid should render the TopTenApp Page 1`] = `ShallowWrapper {}`;

exports[`ImageGrid should render the image grid page 1`] = `ShallowWrapper {}`;
